src/averageAges.js

```diff
@@ -55,7 +89,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childs = people
+    .filter(
+      women => people.some(person => person.name === women.mother)
+    );
+  const childsAges = childs
+    .map(
+      women =>
+        women.born - people.find(person => person.name === women.mother).born
+    );
+  const childsAgesDifference = childsAges
+    .reduce(
+      (sum, years) => sum + years) / childs.length;
+
+  const sons = people
+    .filter(
+      kid =>
+        people.some(person => person.name === kid.mother) && kid.sex === 'm'
```

Comment:
```
      kid =>
        people.some(person => person.name === kid.mother) && kid.sex === 'm'
```
так делать не нужно, обычно делаеют так 
```suggestion
      kid => (
        people.some(person => person.name === kid.mother) && kid.sex === 'm'
      )
```